client/state/account-recovery/reset/actions.js

```diff
@@ -51,3 +54,25 @@ export const updatePasswordResetUserData = ( userData ) => ( {
 	type: ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA,
 	userData,
 } );
+
+export const requestPasswordResetSuccess = () => ( {
+	type: ACCOUNT_RECOVERY_RESET_REQUEST_SUCCESS,
+} );
+
+export const requestPasswordResetError = ( error ) => ( {
+	type: ACCOUNT_RECOVERY_RESET_REQUEST_ERROR,
+	error,
+} );
+
+export const requestPasswordReset = ( request ) => ( dispatch ) => {
```

Comment:
I think we're moving away from using action thunks and shifting towards doing this kind of thing in middleware. I wont be around much tonight to help out on that front but maybe @jordwest can give some pointers there?

I can take another stab at this tomorrow if needed